src/System.Device.Gpio.Tests/GpioControllerSoftwareTests.cs

```diff
@@ -0,0 +1,307 @@
+            // Was this the intended use case?
+            Assert.Throws<InvalidOperationException>(() =>
+            {
+                Span<PinValuePair> wrongArg = stackalloc PinValuePair[2];
+                ctrl.Read(wrongArg);
+            });
+
+            Span<PinValuePair> toread = stackalloc PinValuePair[2];
+            toread[0] = new PinValuePair(1, PinValue.Low);
+            toread[1] = new PinValuePair(2, PinValue.Low);
+            ctrl.Read(toread);
+            Assert.Equal(1, toread[0].PinNumber);
+            Assert.Equal(2, toread[1].PinNumber);
+            Assert.Equal(PinValue.Low, toread[0].PinValue);
+            Assert.Equal(PinValue.High, toread[1].PinValue);
+            ctrl.ClosePin(1);
+            ctrl.ClosePin(2);
+            Assert.False(ctrl.IsPinOpen(1));
+        }
+
+        [Fact]
+        public void WaitForEventAsyncFail()
+        {
+            var ctrl = new GpioController(PinNumberingScheme.Logical, _mockedGpioDriver.Object);
+            _mockedGpioDriver.Setup(x => x.OpenPinEx(1));
+            _mockedGpioDriver.Setup(x => x.IsPinModeSupportedEx(1, PinMode.Input)).Returns(true);
+            _mockedGpioDriver.Setup(x => x.WaitForEventEx(1, PinEventTypes.Rising | PinEventTypes.Falling, It.IsAny<CancellationToken>()))
+                .Returns(new WaitForEventResult()
+                {
+                    EventTypes = PinEventTypes.None, TimedOut = true
+                });
+            Assert.NotNull(ctrl);
+            ctrl.OpenPin(1, PinMode.Input);
+
+            var task = ctrl.WaitForEventAsync(1, PinEventTypes.Falling | PinEventTypes.Rising, TimeSpan.FromSeconds(0.01)).AsTask();
+            task.Wait(CancellationToken.None);
+            Assert.True(task.IsCompleted);
+            Assert.Null(task.Exception);
+            Assert.True(task.Result.TimedOut);
+            Assert.Equal(PinEventTypes.None, task.Result.EventTypes);
+        }
+
+        [Fact]
+        public void WaitForEventSuccess()
+        {
+            var ctrl = new GpioController(PinNumberingScheme.Logical, _mockedGpioDriver.Object);
+            _mockedGpioDriver.Setup(x => x.OpenPinEx(1));
+            _mockedGpioDriver.Setup(x => x.IsPinModeSupportedEx(1, PinMode.Input)).Returns(true);
+            _mockedGpioDriver.Setup(x => x.WaitForEventEx(1, PinEventTypes.Rising | PinEventTypes.Falling, It.IsAny<CancellationToken>()))
+                .Returns(new WaitForEventResult()
+                {
+                    EventTypes = PinEventTypes.Falling,
+                    TimedOut = false
+                });
+            Assert.NotNull(ctrl);
+            ctrl.OpenPin(1, PinMode.Input);
+
+            var result = ctrl.WaitForEvent(1, PinEventTypes.Falling | PinEventTypes.Rising, TimeSpan.FromSeconds(0.01));
+            Assert.False(result.TimedOut);
+            Assert.Equal(PinEventTypes.Falling, result.EventTypes);
+        }
+
+        // TODO: This is still broken. See #974
+        ////[Fact]
```

Comment:
Only 2 // should be enougth :-) Not blocking, and thanks for adding the comment that this code even if dead should stay here

Reply:
If only using two // you get StyleCop errors, because then the comment must be immediatelly followed by code. And a /// comment on the next function (if any) must not, on the other hand, be preceeded by a // comment. So //// is the only way to properly comment out an entire function.

Reply:
/*perhaps there are other ways*/

Reply:
(I'm fine with quadra comments though)